### DISCUSSION TITLE
How to use latest VISTA-3D in Slicer?

### DISCUSSION THREAD

**Discussion by amitjc** [Project-MONAI/MONAILabel#1741] [Q&A]
Dear Experts,

The latest [VISTA-3D](https://github.com/Project-MONAI/VISTA) has a phenomenal potential to segment organs and tumors [Liver, hepatic masses, hepatic vessels (portal vein, hepatic veins)]. I thank @heyufan1995 et al. for this model. 
However, this currently (as far as I can Understand) can only be executed through commandline. This doesn't allow the point-click interaction with CT scan images as the earlier version [Vista-2.5D](https://github.com/Project-MONAI/VISTA/tree/vista2.5d) offered. This limits the capability of refining the segmentation, using the interactive slicer tools.

Is there a possibility of incorporating the latest VISTA-3D in the slicer MONAILabel plugin?

Look forward to guidance for achieving this through the Slicer plugin.

Thank you.

Best Regards,
Amit.

**Comment by pieper** [comment]
Writing such a module should be straightforward if someone wants to invest the time.

A place to start would be to see if the code can be installed directly in the Slicer python environment.

https://github.com/Project-MONAI/VISTA/tree/main/vista3d

**Reply by amitjc** [reply]
Thank @pieper for prompt reply.

I tried replacing the 2.5D model weight with the vista3d weight and modified [vista_point_2pt5.txt](https://github.com/user-attachments/files/16933564/vista_point_2pt5.txt) but failed to have the plugin operational.

Look forward to further suggestions.

Thanks and Regards,
Amit.

**Reply by pieper** [reply]
> Look forward to further suggestions.

Well, if you are familiar with Python you can review the code and look at how similar Slicer extensions.  If you aren't a programmer you may need to find a collaborator or wait to see if someone implements what you are looking for.
